### DISCUSSION TITLE
Running errors with Start.R scripts

### DISCUSSION THREAD

**Discussion by KeLiu7** [magpiemodel/magpie#407] [General]
Hi Jan,

When I was running the Start.R sripts, I got the error below. Could you please guide me how to solve this issue? Thanks

Error in system("git show -s --format=%ci", intern = TRUE) : 
  'git' not found
...model unlocked!
Saving timePrepareStart and timePrepareEnd to runstatistics.rda
Warning message:
package 'magclass' was built under R version 4.0.5 
Loading required package: gdx
Loading required package: gdxrrw

Starting MAgPIE...
[1] 2
Error: MAgPIE model run did not finish properly (fulldata.gdx is missing). Please check full.lst for further information!
Execution halted

**Comment by tscheypidi** [comment]
looks similar to the `gams not found` issue which shows up if gams is not part of the `PATH` variable. Please make sure that also `git` is found on the command line by adding it to the `PATH` variable. 

In case that `git` is not the problem here (I am unsure whether the reported error is a critical one or whether the model just continues to run) you should also have a look at the `full.lst` as the model here suggests. This might give you additional insights into what went wrong.

**Reply by KeLiu7** [reply]
> odel here suggests. This might give you additional insights into w

Thanks Jan. Could you please guide me on how to read the content of full.lst? e.g. which lines generally report the errors? 

I added the git to the path  variable and run again but get different errors this time:

> system("git config core.hooksPath .githooks")
fatal: not in a git directory
[1] 128

All codeCheck tests passed!
Error in as.POSIXlt.character(x, tz, ...) : 
  character string is not in a standard unambiguous format
In addition: Warning messages:
1: In system("git rev-parse HEAD", intern = TRUE) :
  running command 'git rev-parse HEAD' had status 128
2: In system("git status", intern = TRUE) :
  running command 'git status' had status 128
3: In system("git rev-parse HEAD", intern = TRUE) :
  running command 'git rev-parse HEAD' had status 128
4: In system("git show -s --format=%ci", intern = TRUE) :
  running command 'git show -s --format=%ci' had status 128
...model unlocked!
Saving timePrepareStart and timePrepareEnd to runstatistics.rda
Warning message:
package 'magclass' was built under R version 4.0.5 
Loading required package: gdx
Loading required package: gdxrrw

Starting MAgPIE...
[1] 2
Error: MAgPIE model run did not finish properly (fulldata.gdx is missing). Please check full.lst for further information!
Execution halted
Warning message:
In system("git status", intern = TRUE) :
  running command 'git status' had status 128

**Comment by tscheypidi** [comment]
How did you download the model? Did you clone it via git as recommended or did you just download it without git support? You should clone it and then you should normally have a git directory.

**Reply by KeLiu7** [reply]
> How did you download the model? Did you clone it via git as recommended or did you just download it without git support? You should clone it and then you should normally have a git directory.

Thanks Jan. I cloned it via Git as suggested but got another new error. it said "fulldata.gdx" is missing. Do you know how to solve it?  I have attached my full.lst for your reference. Many thanks!
[full.zip](https://github.com/magpiemodel/magpie/files/9050383/full.zip)


Starting MAgPIE...
[1] 2
Error: MAgPIE model run did not finish properly (fulldata.gdx is missing). Please check full.lst for further information!
Execution halted

**Reply by tscheypidi** [reply]
In the `full.lst` you need to look for lines where you have a `$` followed by a number. This is the way how compilation errors are being shown in gams. In your case the error is the following:

```
317202  set    kfo_rd(kfo) Ruminant meat and dairy food products / livst_rum,livst_milk /;
****                                                            $194
**** 194  Symbol redefined - a second data statement for the same symbol
****         or a data statement after an assignment
```
So it looks like the code is somehow inconsistent. If you have changed something in the gams code, then you should revert it and try a default run. If you haven't changed anything, try to pull the most recent changes of the develop branch and try again. Normally, all our official releases as well as all versions from the develop branch should run without compilation error when default settings are being used, but I guess there exceptions to that rule where something went wrong.

**Reply by KeLiu7** [reply]
Thanks Jan. I did not change anything in the gams code. I'm not very clear what you mean here "try to pull the most recent changes of the develop branch". Did you mean should I redownload the input files and try again? if yes, I did so and got the same error.

P.S. Will be the error related my GAMS version or license stuff?

**Reply by tscheypidi** [reply]
The model is under continuous develop and you can via git retrieve the most recent changes in the code without having to re-download the whole model code. The command for this is `git pull`. I would recommend you to have a look at one of the various git tutorials out there to get familiar with the git syntax.

The error does not seem to be related to the GAMS license, the error message would otherwise say so.

**Reply by KeLiu7** [reply]
Hi Jan, I use the command you suggested to retrieve but the error is still there. Do you have any other suggestions?

Would it be related to my old GAMS version (GAMS Studio 0.9.2, GAMS Distribution 25.1.3). The codes might be different from the latest?

Sorry for disturbing you too much..

#######################################
keliu@Ke-Liu MINGW64 /f/magpie (master)
$ git pull
Already up to date.

**Reply by tscheypidi** [reply]
what shows `git status` if you run it in the magpie main folder?

**Reply by KeLiu7** [reply]
Hi Jan, below are what is shown after running git status in the magpie main folder. Should I use git add or git store in this case?

keliu@Ke-Liu MINGW64 /f/magpie (master)
$ git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   main.gms
        modified:   modules/30_crop/penalty_apr22/input.gms
        modified:   modules/30_crop/rotation_apr22/input.gms
        modified:   modules/38_factor_costs/module.gms
        modified:   modules/42_water_demand/agr_sector_aug13/declarations.gms
        modified:   modules/42_water_demand/agr_sector_aug13/postsolve.gms
        modified:   modules/42_water_demand/all_sectors_aug13/declarations.gms
        modified:   modules/44_biodiversity/bv_btc_mar21/input.gms
        modified:   start.R

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        .RData

no changes added to commit (use "git add" and/or "git commit -a")

**Reply by tscheypidi** [reply]
you have changes in the `start.R` as well as in the factor costs module where I don't know where they are coming from. Perhaps you run `git stash` to remove all changes in the code and try to run the model again. My guess is that one of these changes introduced the problem you are getting.

**Reply by KeLiu7** [reply]
Thanks Jan. I tried both git stash and git restore to remove those changes but got the same error. I then went back to run git status and those changes occurred again. Do you know why this happened?

**Reply by tscheypidi** [reply]
some files are being updated from the start scripts, but `start.R` should not change and I see different files modified on my system when I run the most recent master.

Perhaps, to avoid issues with git, you could make another test, where you download a clean copy of the model and start it via `Rscript start.R`

**Reply by KeLiu7** [reply]
Hi Jan, I downloaded the model via git and installed it in another new folder but the error is still there... Any other solution to solve this issue?


Starting MAgPIE...
[1] 2
Error: MAgPIE model run did not finish properly (fulldata.gdx is missing). Please check full.lst for further information!
Execution halted

**Reply by tscheypidi** [reply]
other option might be that your R packages are partly outdated. Please open R and make sure that you have these two package repositories in your repos variable:

```
> getOption("repos")
                                   CRAN                                     pik 
            "https://cran.rstudio.com/" "https://rse.pik-potsdam.de/r/packages" 
```

if you have both, you can run `update.packages()` to make sure that all packages are up-to-date. After that you could try to run the model again

**Reply by KeLiu7** [reply]
Hi Jan, I updated all required packages in the Start. R script and run again. I got some new errors below. Could you please check what's wrong here? Thanks!

Loading required package: gdx
Loading required package: gdxrrw
Error loading the GDX API from gamsSysDir=C:\GAMS\win64\25.1
Could not load symbol 'CgdxOpenReadEx': gdxdclib: gdxOpenReadEx cannot be found in the library.
Environment variable R_GAMS_SYSDIR not set: no GDX API there
Error loading the GDX API via the default shared library search mechanism
Could not load symbol 'CgdxOpenReadEx': gdxdclib: gdxOpenReadEx cannot be found in the library.
PATH = C:\rtools42/x86_64-w64-mingw32.static.posix/bin;C:\rtools42/usr/bin;\usr\bin;C:\rtools42/x86_64-w64-mingw32.static.posix/bin;C:\rtools42/usr/bin;\usr\bin;C:\Program Files\R\R-4.2.1\bin\x64;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\GAMS\win64\25.1;C:\GAMS\win64\25.1\gbin;C:\Program Files\dotnet\;C:\;C:\rtools42\mingw64\bin;C:\rtools42\usr\bin;C:\Program Files\Java\jdk-18.0.1.1\bin;C:\Program Files\Java\jre1.8.0_333\bin;C:\GAMS\win64\25.1;C:\Program Files\Git\cmd;C:\Program Files\Git;C:\Users\keliu\AppData\Local\Microsoft\WindowsApps;C:\GAMS\win64\25.1;C:\Users\keliu\AppData\Local\Programs\MiKTeX\miktex\bin\x64\;C:\Program Files\R\R-4.0.4\bin\x64;C:\Program Files\Git;
The GDX library has not been loaded
Error loading the GDX API from gamsSysDir=C:\GAMS\win64\25.1\gbin
Could not load shared library C:\GAMS\win64\25.1\gbin\gdxdclib64.dll: The specified module could not be found.
Environment variable R_GAMS_SYSDIR not set: no GDX API there
Error loading the GDX API via the default shared library search mechanism
Could not load symbol 'CgdxOpenReadEx': gdxdclib: gdxOpenReadEx cannot be found in the library.
PATH = C:\rtools42/x86_64-w64-mingw32.static.posix/bin;C:\rtools42/usr/bin;\usr\bin;C:\rtools42/x86_64-w64-mingw32.static.posix/bin;C:\rtools42/usr/bin;\usr\bin;C:\Program Files\R\R-4.2.1\bin\x64;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\GAMS\win64\25.1;C:\GAMS\win64\25.1\gbin;C:\Program Files\dotnet\;C:\;C:\rtools42\mingw64\bin;C:\rtools42\usr\bin;C:\Program Files\Java\jdk-18.0.1.1\bin;C:\Program Files\Java\jre1.8.0_333\bin;C:\GAMS\win64\25.1;C:\Program Files\Git\cmd;C:\Program Files\Git;C:\Users\keliu\AppData\Local\Microsoft\WindowsApps;C:\GAMS\win64\25.1;C:\Users\keliu\AppData\Local\Programs\MiKTeX\miktex\bin\x64\;C:\Program Files\R\R-4.0.4\bin\x64;C:\Program Files\Git;
The GDX library has not been loaded
Error loading the GDX API from gamsSysDir=C:\GAMS\win64\25.1
Could not load symbol 'CgdxOpenReadEx': gdxdclib: gdxOpenReadEx cannot be found in the library.
Environment variable R_GAMS_SYSDIR not set: no GDX API there
Error loading the GDX API via the default shared library search mechanism
Could not load symbol 'CgdxOpenReadEx': gdxdclib: gdxOpenReadEx cannot be found in the library.
PATH = C:\rtools42/x86_64-w64-mingw32.static.posix/bin;C:\rtools42/usr/bin;\usr\bin;C:\rtools42/x86_64-w64-mingw32.static.posix/bin;C:\rtools42/usr/bin;\usr\bin;C:\Program Files\R\R-4.2.1\bin\x64;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\GAMS\win64\25.1;C:\GAMS\win64\25.1\gbin;C:\Program Files\dotnet\;C:\;C:\rtools42\mingw64\bin;C:\rtools42\usr\bin;C:\Program Files\Java\jdk-18.0.1.1\bin;C:\Program Files\Java\jre1.8.0_333\bin;C:\GAMS\win64\25.1;C:\Program Files\Git\cmd;C:\Program Files\Git;C:\Users\keliu\AppData\Local\Microsoft\WindowsApps;C:\GAMS\win64\25.1;C:\Users\keliu\AppData\Local\Programs\MiKTeX\miktex\bin\x64\;C:\Program Files\R\R-4.0.4\bin\x64;C:\Program Files\Git;
The GDX library has not been loaded
Error loading the GDX API from gamsSysDir=C:\GAMS\win64\25.1
Could not load symbol 'CgdxOpenReadEx': gdxdclib: gdxOpenReadEx cannot be found in the library.
Environment variable R_GAMS_SYSDIR not set: no GDX API there
Error loading the GDX API via the default shared library search mechanism
Could not load symbol 'CgdxOpenReadEx': gdxdclib: gdxOpenReadEx cannot be found in the library.
PATH = C:\rtools42/x86_64-w64-mingw32.static.posix/bin;C:\rtools42/usr/bin;\usr\bin;C:\rtools42/x86_64-w64-mingw32.static.posix/bin;C:\rtools42/usr/bin;\usr\bin;C:\Program Files\R\R-4.2.1\bin\x64;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\GAMS\win64\25.1;C:\GAMS\win64\25.1\gbin;C:\Program Files\dotnet\;C:\;C:\rtools42\mingw64\bin;C:\rtools42\usr\bin;C:\Program Files\Java\jdk-18.0.1.1\bin;C:\Program Files\Java\jre1.8.0_333\bin;C:\GAMS\win64\25.1;C:\Program Files\Git\cmd;C:\Program Files\Git;C:\Users\keliu\AppData\Local\Microsoft\WindowsApps;C:\GAMS\win64\25.1;C:\Users\keliu\AppData\Local\Programs\MiKTeX\miktex\bin\x64\;C:\Program Files\R\R-4.0.4\bin\x64;C:\Program Files\Git;
The GDX library has not been loaded
Error loading the GDX API from gamsSysDir=C:\GAMS\win64\25.1
Could not load symbol 'CgdxOpenReadEx': gdxdclib: gdxOpenReadEx cannot be found in the library.
Environment variable R_GAMS_SYSDIR not set: no GDX API there
Error loading the GDX API via the default shared library search mechanism
Could not load symbol 'CgdxOpenReadEx': gdxdclib: gdxOpenReadEx cannot be found in the library.
PATH = C:\rtools42/x86_64-w64-mingw32.static.posix/bin;C:\rtools42/usr/bin;\usr\bin;C:\rtools42/x86_64-w64-mingw32.static.posix/bin;C:\rtools42/usr/bin;\usr\bin;C:\Program Files\R\R-4.2.1\bin\x64;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\GAMS\win64\25.1;C:\GAMS\win64\25.1\gbin;C:\Program Files\dotnet\;C:\;C:\rtools42\mingw64\bin;C:\rtools42\usr\bin;C:\Program Files\Java\jdk-18.0.1.1\bin;C:\Program Files\Java\jre1.8.0_333\bin;C:\GAMS\win64\25.1;C:\Program Files\Git\cmd;C:\Program Files\Git;C:\Users\keliu\AppData\Local\Microsoft\WindowsApps;C:\GAMS\win64\25.1;C:\Users\keliu\AppData\Local\Programs\MiKTeX\miktex\bin\x64\;C:\Program Files\R\R-4.0.4\bin\x64;C:\Program Files\Git;
The GDX library has not been loaded
Error loading the GDX API from gamsSysDir=C:\GAMS\win64\25.1\gbin
Could not load shared library C:\GAMS\win64\25.1\gbin\gdxdclib64.dll: The specified module could not be found.
Environment variable R_GAMS_SYSDIR not set: no GDX API there
Error loading the GDX API via the default shared library search mechanism
Could not load symbol 'CgdxOpenReadEx': gdxdclib: gdxOpenReadEx cannot be found in the library.
PATH = C:\rtools42/x86_64-w64-mingw32.static.posix/bin;C:\rtools42/usr/bin;\usr\bin;C:\rtools42/x86_64-w64-mingw32.static.posix/bin;C:\rtools42/usr/bin;\usr\bin;C:\Program Files\R\R-4.2.1\bin\x64;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\GAMS\win64\25.1;C:\GAMS\win64\25.1\gbin;C:\Program Files\dotnet\;C:\;C:\rtools42\mingw64\bin;C:\rtools42\usr\bin;C:\Program Files\Java\jdk-18.0.1.1\bin;C:\Program Files\Java\jre1.8.0_333\bin;C:\GAMS\win64\25.1;C:\Program Files\Git\cmd;C:\Program Files\Git;C:\Users\keliu\AppData\Local\Microsoft\WindowsApps;C:\GAMS\win64\25.1;C:\Users\keliu\AppData\Local\Programs\MiKTeX\miktex\bin\x64\;C:\Program Files\R\R-4.0.4\bin\x64;C:\Program Files\Git;
The GDX library has not been loaded
Error loading the GDX API from gamsSysDir=C:\GAMS\win64\25.1
Could not load symbol 'CgdxOpenReadEx': gdxdclib: gdxOpenReadEx cannot be found in the library.
Environment variable R_GAMS_SYSDIR not set: no GDX API there
Error loading the GDX API via the default shared library search mechanism
Could not load symbol 'CgdxOpenReadEx': gdxdclib: gdxOpenReadEx cannot be found in the library.
PATH = C:\rtools42/x86_64-w64-mingw32.static.posix/bin;C:\rtools42/usr/bin;\usr\bin;C:\rtools42/x86_64-w64-mingw32.static.posix/bin;C:\rtools42/usr/bin;\usr\bin;C:\Program Files\R\R-4.2.1\bin\x64;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\GAMS\win64\25.1;C:\GAMS\win64\25.1\gbin;C:\Program Files\dotnet\;C:\;C:\rtools42\mingw64\bin;C:\rtools42\usr\bin;C:\Program Files\Java\jdk-18.0.1.1\bin;C:\Program Files\Java\jre1.8.0_333\bin;C:\GAMS\win64\25.1;C:\Program Files\Git\cmd;C:\Program Files\Git;C:\Users\keliu\AppData\Local\Microsoft\WindowsApps;C:\GAMS\win64\25.1;C:\Users\keliu\AppData\Local\Programs\MiKTeX\miktex\bin\x64\;C:\Program Files\R\R-4.0.4\bin\x64;C:\Program Files\Git;
The GDX library has not been loaded
Error loading the GDX API from gamsSysDir=C:\GAMS\win64\25.1
Could not load symbol 'CgdxOpenReadEx': gdxdclib: gdxOpenReadEx cannot be found in the library.
Environment variable R_GAMS_SYSDIR not set: no GDX API there
Error loading the GDX API via the default shared library search mechanism
Could not load symbol 'CgdxOpenReadEx': gdxdclib: gdxOpenReadEx cannot be found in the library.
PATH = C:\rtools42/x86_64-w64-mingw32.static.posix/bin;C:\rtools42/usr/bin;\usr\bin;C:\rtools42/x86_64-w64-mingw32.static.posix/bin;C:\rtools42/usr/bin;\usr\bin;C:\Program Files\R\R-4.2.1\bin\x64;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\GAMS\win64\25.1;C:\GAMS\win64\25.1\gbin;C:\Program Files\dotnet\;C:\;C:\rtools42\mingw64\bin;C:\rtools42\usr\bin;C:\Program Files\Java\jdk-18.0.1.1\bin;C:\Program Files\Java\jre1.8.0_333\bin;C:\GAMS\win64\25.1;C:\Program Files\Git\cmd;C:\Program Files\Git;C:\Users\keliu\AppData\Local\Microsoft\WindowsApps;C:\GAMS\win64\25.1;C:\Users\keliu\AppData\Local\Programs\MiKTeX\miktex\bin\x64\;C:\Program Files\R\R-4.0.4\bin\x64;C:\Program Files\Git;
The GDX library has not been loaded

Starting MAgPIE...
[1] 2
Error: MAgPIE model run did not finish properly (fulldata.gdx is missing). Please check full.lst for further information!
Execution halted

**Reply by tscheypidi** [reply]
something seems to be off with your package/GAMS installation as the `gdxrrw` R package does not seem to be able to properly connect to the GAMS libraries. Perhaps go back to the installation guidelines @ https://github.com/magpiemodel/magpie#how-to-install and redo all these steps.

**Comment by danielemosso** [comment]
Hi, I am also getting the same error when trying to run MAgPIE. 

In a previous installation, I entered in some troubles with gdxrrw package and Mingw. 

Now, I have done evertything again from scratch, not encountering the previous issue.

Neverthless, when i run the model, i get:


Error: The execution of the MAgPIE model was not completed correctly (fulldata.gdx file missing). Check full.lst for further information!
Execution aborted
2024-04-17 17:03:53.586896: model unlocked.
Warning messages:
1: In untar2(tarfile, files, list, exdir, restore_times) :
 use of extended pax headers
2: In untar2(tarfile, files, list, exdir, restore_times) :
 using pax extended headers
3: In untar2(tarfile, files, list, exdir, restore_times) :
 using pax extended headers
4: In untar2(tarfile, files, list, exdir, restore_times) :
 using pax extended headers

What can be the issue? It may be realted to mingw or in general at the interaction with a GCC compiler? I do not have any compiler appearing in my environment variables for example..

**Reply by danielemosso** [reply]
Maybe solved, navigating to the log files i found:

--- Job full.gms Start 04/17/24 17:03:53 46.4.0 2cbe6e3b WEX-WEI x86 64bit/MS Windows
--- Applying:
    C:\GAMS\46\gmsprmNT.txt
--- GAMS Parameters defined
    Input C:\Users\Daniele\magpie\output\default_2024-04-17_16.59.36\full.gms
    ScrDir C:\Users\Daniele\magpie\output\default_2024-04-17_16.59.36\225a\
    SysDir C:\GAMS\46\
    LogOption 2
    LogFile C:\Users\Daniele\magpie\output\default_2024-04-17_16.59.36\full.log
    ErrMsg 1
*** License file too old for this version of GAMS.
*** Maintenance expired 1200 days ago.
*** More than 30 days since expiration, sorry...
*** Run an older GAMS system or renew the license
*** Status: Terminated due to a licensing error
***         License file: C:\GAMS\46\gamslice.txt
***         Inspect listing file for more information.
--- Job full.gms Stop 04/17/24 17:03:53 elapsed 0:00:00.004

I'll try with an older version of GAMS hoping it works

**Reply by tscheypidi** [reply]
yes, moving to an older version should work in this case.

**Reply by danielemosso** [reply]
Thank you! Unfortunately, I do not have access to other verions since my university do not have a GAMS license and my student one is not enough. What I have just tried is to use NEOS server to run CONOPT by uploading my GAMS main file. I faced the following issue:

*************************************************************

   NEOS Server Version 6.0
   Job#     : 14115387
   Password : DJACIwpd
   User     : danielemosso
   Solver   : nco:CONOPT:GAMS
   Start    : 2024-04-18 07:58:01
   End      : 2024-04-18 07:58:04
   Host     : prod-sub-1.neos-server.org

   Disclaimer:

   This information is provided without any express or
   implied warranty. In particular, there is no warranty
   of any kind concerning the fitness of this
   information for any particular purpose.

   Announcements:
*************************************************************
Executed on prod-exec-6.neos-server.org
GAMS 45.7.0  64fbf3ce Jan 18, 2024          LEX-LEG x86 64bit/Linux - 04/18/24 07:58:04 Page 1
magpie
C o m p i l a t i o n


 305  $include "./core/macros.gms"
****                             $282
**** 282  Unable to open include file
****         Hint: Does the filename need to be quoted?
 309  $include "./core/sets.gms"
****                           $282
**** 282  Unable to open include file
****         Hint: Does the filename need to be quoted?
 310  $batinclude "./modules/include.gms" sets
****                                     $282
**** 282  Unable to open include file
****         Hint: Does the filename need to be quoted?
 314  $include "./core/declarations.gms"
****                                   $282
**** 282  Unable to open include file
****         Hint: Does the filename need to be quoted?
 315  $batinclude "./modules/include.gms" declarations
****                                     $282
**** 282  Unable to open include file
****         Hint: Does the filename need to be quoted?
 319  $batinclude "./modules/include.gms" input
****                                     $282
**** 282  Unable to open include file
****         Hint: Does the filename need to be quoted?
 323  $batinclude "./modules/include.gms" equations
****                                     $282
**** 282  Unable to open include file
****         Hint: Does the filename need to be quoted?
 327  model magpie / all - m15_food_demand /;
****                                     $140,233
**** 140  Unknown symbol
**** 233  Empty model list
****        no legal data between / / or 'all' without any equations
 339  $batinclude "./modules/include.gms" scaling
****                                     $282
**** 282  Unable to open include file
****         Hint: Does the filename need to be quoted?

**** 10 ERROR(S)   0 WARNING(S)


COMPILATION TIME     =        0.001 SECONDS      2 MB  45.7.0 64fbf3ce LEX-LEG


USER: NEOS Server License prod-exec-6.neos-server.orgS231116/0001AB-GEN
      mac@d0:94:66:89:89:0f                                     DCE1890
      License for teaching and research at degree granting institutions


**** FILE SUMMARY

Input      /var/lib/condor/execute/dir_1733557/gamsexec/MODEL.gms
Output     /var/lib/condor/execute/dir_1733557/gamsexec/solve.lst

**** USER ERROR(S) ENCOUNTERED


It seems to me that the main file call other files, such as macro,sets and other stuff.

Unfortunately, on the NEOS server i can upload just a single file. 

Do I have any chance in using the solver? Or without a local GAMS and CONOPT license i cannot ?

Best,

Daniele

**Reply by tscheypidi** [reply]
We also did some tests with the NEOS server and unfortunately were not able to find a working solution. The issue of uploading a singe gams file can be solved by starting a MAgPIE run locally. It will then produce a single full.gms file in the output folder containing the whole GAMS model for that specific parametrization. Unfortunately, the resulting gms-filler is bigger than what NEOS allows to be uploaded.

To my knowledge there is currently no workaround available and you need a GAMS + CONOPT license to run MAgPIE. Sorry for that.

**Comment by DCCAISSON** [comment]
Dear MAgPIE Development Team,

I am currently facing an same issue that seems to be related to the problem described in Issue #407 (Running errors with Start.R scripts). Below is the specific error information I encountered while trying to run the MAgPIE model: 

GAMS 44.4.0   Copyright (C) 1987-2023 GAMS Development. All rights reserved
--- Starting compilation
--- full.gms(419585) 276 Mb 1 Error
*** Error 409 in C:\Users\lenovo\Desktop\magpie\output\default_2022-11-14_21.13.06\full.gms
    Unrecognizable item - skip to find a new statement
      looking for a ';' or a key word to get started again
--- full.gms(1376398) 279 Mb 2 Errors
*** Error 257 in C:\Users\lenovo\Desktop\magpie\output\default_2022-11-14_21.13.06\full.gms
    Solve statement not checked because of previous errors
--- full.gms(1376404) 279 Mb 3 Errors
*** Error 257 in C:\Users\lenovo\Desktop\magpie\output\default_2022-11-14_21.13.06\full.gms
    Solve statement not checked because of previous errors
--- full.gms(1376441) 279 Mb 4 Errors
*** Error 141 in C:\Users\lenovo\Desktop\magpie\output\default_2022-11-14_21.13.06\full.gms
    Symbol declared but no values have been assigned. Check for missing
       data definition, assignment, data loading or implicit assignment
       via a solve statement. Suppress with $onImplicitAssign.
       A wild shot: You may have spurious commas in the explanatory
       text of a declaration. Check symbol reference list.
--- full.gms(1376449) 279 Mb 5 Errors
*** Error 141 in C:\Users\lenovo\Desktop\magpie\output\default_2022-11-14_21.13.06\full.gms
    Symbol declared but no values have been assigned. Check for missing
       data definition, assignment, data loading or implicit assignment
       via a solve statement. Suppress with $onImplicitAssign.
       A wild shot: You may have spurious commas in the explanatory
       text of a declaration. Check symbol reference list.

I have reviewed the relevant sections of the GAMS model and tried debugging based on the reported issues. However, despite ensuring that most of the scripts appear to be syntactically correct, these errors persist. Specifically:

Error 409 indicates that an unrecognizable item was found, likely due to a missing semicolon or keyword.
Error 257 prevents checking Solve statements due to previous unresolved errors.
Error 141 reports that symbols were declared but not assigned any values, suggesting possible issues with data assignment or implicit assignment through a Solve statement.
I have verified the model data loading process, yet these errors still occur.

Could you please provide guidance on resolving these errors? Any insights from your development experience or advice on troubleshooting would be greatly appreciated.

Thank you in advance for your help.

Best regards,
DCCAISSON

**Reply by jdrtommey** [reply]
Hi DCCAISSON,

Sorry you are having difficulties with running the model. Could you possibly give us more information about your system configuration, for example what version of MAgPIE you are using and what input data? Also, the folder being used shows a fairly old timestamp `2022-11-14` could perhaps indicate there are some compatibility issues if certain parts of the model are not in sync?

**Reply by DCCAISSON** [reply]
**### Certainly, I would be happy to provide further information. Below are the data usage details for the model operation; I have utilized the default dataset provided by Magpie for the execution.**

Used data set: rev4.109_h12_magpie.tgz
md5sum: NA
Repository: https://rse.pik-potsdam.de/data/magpie/public

Used data set: rev4.109_h12_fd712c0b_cellularmagpie_c200_MRI-ESM2-0-ssp370_lpjml-8e6c5eb1.tgz
md5sum: NA
Repository: https://rse.pik-potsdam.de/data/magpie/public

Used data set: rev4.109_h12_validation.tgz
md5sum: NA
Repository: https://rse.pik-potsdam.de/data/magpie/public

Used data set: additional_data_rev4.51.tgz
md5sum: NA
Repository: https://rse.pik-potsdam.de/data/magpie/public

Used data set: calibration_H12_26Mar24.tgz
md5sum: NA
Repository: https://rse.pik-potsdam.de/data/magpie/public

Low resolution: c200
High resolution: 0.5

Total number of cells: 200

Number of cells per region:
  CAZ  CHA  EUR  IND  JPN  LAM  MEA  NEU  OAS  REF  SSA  USA
   14   23   10    7    4   26   21    9   16   23   32   15

Regionscode: 62eff8f7

Regions data revision: 4.109

lpj2magpie settings:
* LPJmL data: MRI-ESM2-0:ssp370
* Revision: 4.109

aggregation settings:
* Input resolution: 0.5
* Output resolution: c200
* Regionscode: 62eff8f7
* Number of clusters per region:
  CAZ  CHA  EUR  IND  JPN  LAM  MEA  NEU  OAS  REF  SSA  USA
   14   23   10    7    4   26   21    9   16   23   32   15
* Call: withCallingHandlers(expr, message = messageHandler, warning = warningHandler,     error = errorHandler)


Last modification (input data): Sat Sep 21 16:34:35 2024


**Additionally, I am using the latest version of Magpie, which is version 4.8.1.**

**Furthermore, I am also considering whether the issue might be related to the timestamp. If the incompatibility is caused by the timestamp, it may be a problem that I am temporarily unable to resolve on my own.**
